File: java/src/privileges/Comment.java
package privileges;

import decorator.PrivilegeDecorator;
import decorator.User;

public class Comment extends PrivilegeDecorator{
	User user;
	
	public Comment(User user) {
		this.user = user;
	}

	@Override
	public String getPrivileges() {
		return user.getPrivileges() + " comment";
	}
	
	@Override
	public String getName() {
		return user.getName();
	}
}

File: csharp/decorator_practice/decorator_practice/main/privileges/Comment.cs
using decorator_practice.main.decorator;
using System;
using System.Collections.Generic;
using System.Text;

namespace decorator_practice.main.privileges
{
    public class Comment : PrivilegeDecorator
    {
        /**
	     * User.
	     */
        private User user;

        /**
	     * @param u constructor.
	     */
        public Comment(User u)
        {
            this.user = u;
        }

        /**
	     * @return approve privilege.
	     */
        override
        public String getPrivileges()
        {
            return user.getPrivileges() + " comment";
        }

        /**
	     * @return name.
	     */
        override
        public String getName()
        {
            return user.getName();
        }
    }
}
